pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -326,10 +328,50 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
 		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add}},
 		// Pods may fail to schedule because the PVC it uses has not yet been created.
 		// This PVC is required to exist to check its access modes.
-		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update},
+			QueueingHintFn: pl.isSchedulableAfterPersistentVolumeClaimChange},
 	}
 }
 
+// isSchedulableAfterPersistentVolumeClaimChange is invoked whenever a PersistentVolumeClaim added or changed, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldPersistentVolumeClaim, newPersistentVolumeClaim, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPersistentVolumeClaimChange: %w", err)
+	}
+
+	if oldPersistentVolumeClaim != nil || newPersistentVolumeClaim.Namespace != pod.Namespace {
+		return framework.QueueSkip, nil
+	}
+
+	pvcs := sets.New[string]()
+
+	for _, volume := range pod.Spec.Volumes {
```

Comment:
We should only check volumes which can be rejected by `volumeRestriction`.
https://github.com/kubernetes/kubernetes/blob/2d8a3ad5728cb6ff00f9721303c18e942ccfbfc8/pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go#L157-L159

I think that means we should only check these 4 types of volumes and pvc with readWriteOncePod accessmode?

Reply:
For any pod, if its pvc is not created, it cannot be scheduled, right? I think this has nothing to do with the type of volume, and whether the pvc is `readWriteOncePod`.

Reply:
The current code will function correctly. However, the purpose of QHInt is to allow the plugin to determine if a previously rejected pod can potentially pass through it again. If the PVC has not been created, it will not be rejected by VolumeRestriction (I'm not sure but maybe volume_binding?). Although adding this check will not make the performance faster, I believe we should maintain consistent behavior.

Reply:
Thank you for your careful review :)